src/hooks/con-use-facet-name-resolution.js

```diff
@@ -21,11 +21,30 @@ export const useFacetNameResolution = (facets, objectStore) => {
 
     const resolveFacetNames = async () => {
       setIsResolving(true);
-      
+
       try {
+        // ALWAYS apply schema label transformations first (synchronous)
+        const resolvedFacetsObj = JSON.parse(JSON.stringify(facets)); // Deep clone
+
+        // Apply schema label transformations immediately
+        const applySchemaTransformations = (facetsObj) => {
+          if (facetsObj['@self']?.schema?.buckets) {
+            facetsObj['@self'].schema.buckets.forEach((bucket) => {
+              if (bucket.label === 'Module') {
+                bucket.label = 'Applicatie';
+              }
+              if (bucket.label === 'Module Versie') {
+                bucket.label = 'Applicatie versie';
+              }
+            });
+          }
+        };
+
+        applySchemaTransformations(resolvedFacetsObj);
```

Comment:
er is geen rede om dit een functie te maken want het wordt maar 1x gebruikt


```suggestion
        if (resolvedFacetsObj['@self']?.schema?.buckets) {
          resolvedFacetsObj['@self'].schema.buckets.forEach((bucket) => {
            if (bucket.label === 'Module') {
              bucket.label = 'Applicatie';
            }
            if (bucket.label === 'Module Versie') {
              bucket.label = 'Applicatie versie';
            }
          });
        }
```